### ISSUE TITLE
Signal can't download MMS

### ISSUE THREAD

**Issue by lazyletucce** [signalapp/Signal-Android#9542]
Hello i'm on android 10 using stable signal and i can't download MMS, it's working with the default SMS app but it's not working in signal (error downloading, touch to to retry), i'm about to give up on signal app but i wanted to report it before in a last hope. any idea when it will get fixed? thanks

**Comment by marissarousseau** [comment]
I'm having the exact same issue. I am using a Pixel 3 running Graphene OS. Not sure if it has to do with not having Google Play Services on my devices or not, but I am constantly having MMS issues with Signal.

**Comment by k3npachi** [comment]
Activate your 4g, not working with wifi

**Comment by marissarousseau** [comment]
> Activate your 4g, not working with wifi

Turns out that my issue was that my firewall app was blocking network access to a system app responsible for sending and receiving MMS. Once I gave it proper network access, my MMS issues vanished. Thank you for your help though!

**Comment by k3npachi** [comment]
Afwall+  ?

**Comment by k3npachi** [comment]
Good app, but needs to be well configured

It happened to me, too.

**Comment by breversa** [comment]
> > Activate your 4g, not working with wifi
> 
> Turns out that my issue was that my firewall app was blocking network access to a system app responsible for sending and receiving MMS. Once I gave it proper network access, my MMS issues vanished. Thank you for your help though!

Can you tell us the name of that app ?

**Comment by marissarousseau** [comment]
> Can you tell us the name of that app ?

AFWall+ is the name. It requires root, and I have since completely stopped using it. I no longer have any of the issues described above.

**Comment by breversa** [comment]
No, I mean the name of the app that you had to allow in AFwall+ to permit sending/receiving MMSs.
I'm a regular AFwall+ user and sometimes stumble upon that issue with some devices.

**Comment by marissarousseau** [comment]
I reinstalled the app to check what the app was called. Doing a search pulled up two different apps. I believe I enabled all network connections for both apps.  
![](https://i.imgur.com/UG87sHQ.jpg)

**Comment by breversa** [comment]
Thank you, that should help for future reference.

**Comment by IrkenEngineer** [comment]
I am currently experiencing the same issue as @lazyletucce, but unfortunately disabling my firewall entirely has not helped me. I am running vanilla Android 10 with the latest Signal app from Google Play on Mint Mobile. I had the same issue for over a year on my old phone, which persisted across multiple custom ROMs, no matter my APN settings. I was hoping a new device with stock Android might solve the issue, but I still am unable to receive MMS messages from non-Signal users.

**Comment by ngrigoriev** [comment]
Indeed, I had the same problem (was actually getting some MMS phishing messages but could not read them ;) ). And in my case it was exactly the second "app" (e.g. "Phone Services"). Allowing it in AFWall+ allowed to download the MMS contents. However, I am not sure if I want to keep this service enabled without knowing what else it is doing.
